testing/endpoint.go

```diff
@@ -117,7 +123,12 @@ func (endpoint *Endpoint) CreateClient() (err error) {
 		if err != nil {
 			return err
 		}
-		clientState = wasmtypes.NewClientState(wasmClientState, endpoint.Chain.Coordinator.CodeHash, height)
+
+		wasmTendermintConfig, ok := endpoint.ClientConfig.(*WasmTendermintConfig)
+		if !ok {
+			return fmt.Errorf("invalid client config type, expected %T, got: %T", (*WasmTendermintConfig)(nil), endpoint.ClientConfig)
+		}
+		clientState = wasmtypes.NewClientState(wasmClientState, wasmTendermintConfig.CodeHash, height)
```

Comment:
That's a really good point. We should really try to avoid that dependency issue.

Reply:
isn't this reminiscent of the 02-client refactor? Couldn't ClientConfig (or some other interface) just have each specific client implement its separate client creation/updating logic for testing purposes? This might just be your suggestion and I'm just being overtly verbose with it.

Reply:
Yeah I think the testing package should probably only have dependencies on things like `exported.ClientState` and `exported.ConsensusState`.

ideally additional client types should be able to be added without changes to the testing package.

It might be nice to have something like

```golang
type ClientCreator interface {
	CreateClientState() exported.ClientState
	CreateConsensusState() exported.ConsensusState
}

func (endpoint *Endpoint) CreateClient(cc ClientCreator) (err error) {
	// ...
	
	clientState := cc.CreateClientState(...)
	consensusState := cc.CreateConsensusState(...)

	msg, err := clienttypes.NewMsgCreateClient(
		clientState, consensusState, endpoint.Chain.SenderAccount.GetAddress().String(),
	)
	require.NoError(endpoint.Chain.TB, err)

	// ...
	
	return nil
}

```

I am a big fan of defining small interfaces near where they are used and passing them as arguments, but this could be achieved in a few different ways. The actual cc could be configured at endpoint creation or passed in like in the code above.

Reply:
will also add that I think this could happen in a follow up, I think the current changes are still an improvement but are also in line with how we treat other clients (e.g. solomachine).

Would be happy to see this as a follow up pr 👍

Reply:
and I thought I was verbose :laughing: